docs/swagger2.yml

```diff
@@ -27,673 +19,414 @@ consumes:
-      $ref: '#/definitions/MemoryResource'
-      $ref: '#/definitions/CPUResource'
-      $ref: '#/definitions/TimeoutResource'
-      $ref: '#/definitions/IdleTimeoutResource'
+        description: "server generated version denoting image_id and configuration version."
       config:
         type: object
-        description: Trigger configuration - overrides application configuration
-        additionalProperties:
-          type: string
-      annotations:
-        type: object
-        description: Trigger annotations - this is a map of annotations attached to this trigger, keys must not exceed 128 bytes and must consist of non-whitespace printable ascii characters, and the seralized representation of individual values must not exeed 512 bytes
-        additionalProperties:
-          type: object
+        description: Function configuration - this is the base configuration
+        properties:
+          mem:
+            type: integer
+            format: uint64
+            description: Max usable memory given to function (MiB).
+          cpu:
+            type: string
+            description: Max usable CPU cores for this trigger. Value in MilliCPUs (eg. 500m) or as floating-point (eg. 0.5)
+          execution_timeout:
+            type: integer
+            default: 30
+            format: int32
+            description: Timeout for executions of a function. Value in Seconds
+          idle_timeout:
+            type: integer
+            default: 30
+            format: int32
+            description: Hot functions idle timeout before container termination. Value in Seconds
+      # TODO(?) annotations?
       created_at:
         type: string
         format: date-time
-        description: Time when trigger was created. Always in UTC.
+        description: Time when function was created. Always in UTC RFC3339.
         readOnly: true
       updated_at:
         type: string
         format: date-time
-        description: Most recent time that trigger was updated. Always in UTC.
+        description: Most recent time that function was updated. Always in UTC RFC3339.
         readOnly: true
+      status:
+        type: string
+        enum: [enabled, disabled]
 
-  App:
+  Trigger:
     type: object
     properties:
       id:
         type: string
-        description: App ID
+        description: UUID
         readOnly: true
       name:
         type: string
-        description: Name of this app. Must be different than the image name. Can ony contain alphanumeric, -, and _.
-      config:
-        type: object
-        description: Application function configuration, applied to all triggers.
-        additionalProperties:
-          type: string
+        description: unique name for this trigger, used to identify this trigger
+      type:
+        type: string
+        description: class of trigger, e.g. schedule, http
+        enum: [http, stream]
```

Comment:
string? we aren't necessarily restricting the domain here and users could 'roll their own' yea? e.g. i want a mongodb trigger (and the ability to list triggers by type may be something we end up with eventually)